src/test/java/lambda/part1/exercise/Exercise3.java

```diff
@@ -0,0 +1,76 @@
+package lambda.part1.exercise;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.FluentIterable;
+import lambda.data.Person;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+@SuppressWarnings({"ConstantConditions", "unused"})
+public class Exercise3 {
+
+    @Test
+    public void sortPersonsByAgeUsingArraysSortExpressionLambda() {
+        Person[] persons = getPersons();
+
+        // TODO использовать Arrays.sort + expression-lambda
+        Arrays.sort(persons, (left, right) -> Integer.compare(left.getAge(), right.getAge()));
+
+        assertArrayEquals(new Person[]{
+            new Person("Иван", "Мельников", 20),
+            new Person("Николай", "Зимов", 30),
+            new Person("Алексей", "Доренко", 40),
+            new Person("Артем", "Зимов", 45)
+        }, persons);
+    }
+
+    @Test
+    public void sortPersonsByLastNameThenFirstNameUsingArraysSortExpressionLambda() {
+        Person[] persons = getPersons();
+
+        // TODO использовать Arrays.sort + statement-lambda
+
+        Arrays.sort(persons, (left, right) -> {
+            int difference = left.getLastName().compareTo(right.getLastName());
+            return difference != 0 ? difference
+                : left.getFirstName().compareTo(right.getFirstName());
+        });
+
+        assertArrayEquals(new Person[]{
+            new Person("Алексей", "Доренко", 40),
+            new Person("Артем", "Зимов", 45),
+            new Person("Николай", "Зимов", 30),
+            new Person("Иван", "Мельников", 20)
+        }, persons);
+    }
+
+    @Test
+    public void findFirstWithAge30UsingGuavaPredicateLambda() {
+        List<Person> persons = Arrays.asList(getPersons());
+
+        // TODO использовать FluentIterable
+
+        Optional<Person> personOptional = FluentIterable.from(persons)
+            .firstMatch(person -> Integer.compare(person.getAge(), 30) == 0);
```

Comment:
No need to use Integer.compare here. A Predicate should return boolean value, not int.

Reply:
I agree that Integer.compare() is redundant here and I'll fix this, but expression returns boolean, because Integer.compare() compares with 0.

Reply:
Right, sorry.